### PR TITLE
Export CAC data to Storepoint

### LINKED ISSUE #8: Export CAC data as Storepoint CSV
Add a rake task to import CAC data in the format Storepoint expects.

### PR DESCRIPTION
**Pull request by ivanoblomov**
Closes #8.

## Goal
Add logic for exporting CAC location-data in Storepoint's format.

## Approach
1. Add:
    1. ActiveModel `CACLocation`, which can become an ActiveRecord later after we implement persistence
    2. instance method `CACLocation#to_storepoint`, which converts the location to a hash of Storepoint fields and values
    3. static utility method `CACLocation.to_storepoint`, which converts an array of locations to hashes of Storepoint fields and values by calling 2
    4. static utility method `CACLocation.all_from_api`, which converts all locations from the API to Storepoint fields and values by calling 3
    5. rake task `:export_cac_as_storepoint`, which invokes 4
2. Cover the meat of the transformation logic, 1ii. We can ignore 1iii and 1iv because they're either trivial or make API calls (which won't work on GitHub's CI).
3. Pass tests.

## Notes
1. I isolated most of the new logic in `lib/models` to facilitate migration to Rails.
2. Note this code doesn't yet:
    1. incorporate the massaging logic @neilberkman is working on
    2. account for Evive data
3. Turned out these files weren't needed, but wanted to commit them to the history just in case we need them later. Will remove in a subsequent PR:
    1. `lib/models/storepoint_location.rb`
    2. `test/fixtures/store_point_location.csv`
    3. `test/models/storepoint_location_test.rb`

### CODE REVIEW

**Comment by ivanoblomov** [comment]
@neilberkman my bad, not sure what Apple/Gmail did there! Copying it here to make sure the image stays intact:

> Neil, wanted to talk through some architectural calls with you, but I didn’t know any good virtual-whiteboard tools off the top of my head.
> 
> Here’s what I was thinking…
> 
> 1 is roughly where we’re at it: Evive (E) data is transformed directly to Storepoint (S).
> Initially, I was thinking my PR (https://github.com/Spark-Bio/covid-storepoint/pull/9) could build the dotted line between CAC (C) and S, but I was wondering whether alternate approaches could be better.
> 
> 2 is one alternative, where E and C data are transformed to an intermediary generic Location (L) that can then be exported to S.
> 
> 3 is a third option, where the generic L is essentially identical to C.
> 
> I think 2 and 3 imply persistence, whereas 1 doesn’t. Merging E and C without a db is difficult though.
> 2 is more flexible, and more complex, than 3.
> 
> Got any preferences—or is there a fourth option I overlooked?
> 
> If you’d rather talk through this synchronously, let me know. I’m free after 2pm ET most days.
> 
> Unknown
>

**Comment by neilberkman** [comment]
> 3 is a third option, where the generic L is essentially identical to C.

I think I prefer this option. The CAC schema is a superset of Evive so I think it works for our purposes.

**Comment by ivanoblomov** [comment]
Sounds good, I was shooting for that!

**Comment by ivanoblomov** [comment]
@neilberkman this is ready for review, but please don't merge yet. I want to branch off a clean-up PR as noted above.

**Comment by ivanoblomov** [comment]
@neilberkman covered and fixed the bugs you found:
1. column alignment
2. `additional_information_for_patients ` for `description`
3. `location_place_of_service_type` for `tags`

I'll open issues for adding:
1. CAC's boolean fields to the description
2. hours parsing

**Comment by neilberkman** [comment]
> @neilberkman covered and fixed the bugs you found

Awesome! I'll take a look tomorrow AM.

File: lib/models/cac_location.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+       location_contact_url_covid_screening_tool
+       location_contact_url_covid_virtual_visit
+       location_contact_url_main location_hours_of_operation
+       location_id location_latitude location_longitude location_name
+       location_place_of_service_type location_specific_testing_criteria
+       location_status raw_data record_id reference_publisher_of_criteria
+       updated_on].freeze
+  CAC_TO_STOREPOINT_MAPPING = {
+    location_name: :name, storepoint_description: :description,
+    location_address_street: :address, location_address_locality: :city,
```
[the remark this answers is not in the document]
Comment (neilberkman) [reply lib/models/cac_location.rb:38]: FYI I opened an issue: https://github.com/codersagainstcovidorg/covid19testing-backend/issues/10

[the remark this answers is not in the document]
Comment (ivanoblomov) [reply lib/models/cac_location.rb:38]: Ugh, can't even imagine what drove that decision.

File: lib/models/cac_location.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  # StorepointLocations.
+  #
+  # @param locations [Array] the CACLocations to convert
+  # @return [Array] StorepointLocations
+  def self.to_storepoint(locations)
+    locations.map(&:to_storepoint)
+  end
+
+  def storepoint_country
+    'USA'
```
[the remark this answers is not in the document]
Comment (ivanoblomov) [reply lib/models/cac_location.rb]: Fixed by 2f2fa5727afa5c703efaef78ed5985de88bc17cc.

**Pull request merged**